daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/X_DFDL_MIL_STD.scala

```diff
@@ -48,6 +48,46 @@ object BitsCharset6BitDFI311DUI002 extends {
   override val requiredBitOrder = BitOrder.LeastSignificantBitFirst
 } with BitsCharsetNonByteSize
 
+object BitsCharset3BitDFI336DUI001 extends {
+  override val name = "X-DFDL-3-BIT-DFI-336-DUI-001"
+  override val bitWidthOfACodeUnit = 3
+  override val decodeString = """12345678"""
+  override val replacementCharCode = 0x0
+  override val requiredBitOrder = BitOrder.LeastSignificantBitFirst
+} with BitsCharsetNonByteSize
+
+object BitsCharset4BitDFI746DUI002 extends {
+  override val name = "X-DFDL-4-BIT-DFI-746-DUI-002"
+  override val bitWidthOfACodeUnit = 4
+  override val decodeString = """ABCDEFGHIJKLMNPQ"""
+  override val replacementCharCode = 0x0
+  override val requiredBitOrder = BitOrder.LeastSignificantBitFirst
+} with BitsCharsetNonByteSize
+
+object BitsCharset3BitDFI746DUI002 extends {
+  override val name = "X-DFDL-3-BIT-DFI-746-DUI-002"
+  override val bitWidthOfACodeUnit = 3
+  override val decodeString = """ABCDEFGH"""
+  override val replacementCharCode = 0x0
+  override val requiredBitOrder = BitOrder.LeastSignificantBitFirst
+} with BitsCharsetNonByteSize
+
+object BitsCharset3BitDFI747DUI001 extends {
+  override val name = "X-DFDL-3-BIT-DFI-747-DUI-001"
```

Comment:
Minor and probably not worth changing, but I've never been a fan of charset names that contain DFI/DUI's. It seems so specific to a single use case of a charset. Granted most of them probably are the only cases, but still seems odd. Especially since encodings *could* be shared by by different DFI/DUIs. For example, maybe DFI 123 DUI 456 has the same encoding. Using this encoding for that DFI/DUI now will look like a bug in the schema, even though it's would be correct. But maybe the solution there is to just add aliases?

I'd maybe rather see something like X-DFDL-TDL-NNNN-M, where NNNN a zero-padded number that we increment for every new TDL encoding we come across, and M is the bit length. So this might be X-DFDL-TDL-0010-3.

Reply:
The notion here is that we pick one of the dfi/dui as the "canonical" usage, and any other dfi/dui that share the encoding would use that same encoding, and the fact that it is shared would not look like a mistake, as much as intention to share the "canonical" one. Lots of things are shared across dfi/duis so this seems an ok course.  Or we could do aliases as you suggest. There are numerous specs that could be re-using these encodings, but all of them are these DFI/DUI-style specs, so I think there is little gained by avoiding the DFI/DUI identification. 

There is an argument to get the dfi/dui notation entirely out of Daffodil, but in doing so we just make it hard for anyone using Daffodil to know if the charset they need is in fact part of daffodil, so we'd end up documenting them as X-DFDL-TDL-NNNN-M is the charset needed for DFI X DUI Y anyway.